docs-markdown/src/helper/Auth.ts

```diff
@@ -16,36 +17,35 @@ export interface TokenResponse {
 	tokenType: string;
 }
 
-interface AppConfig {
-	instance: string;
-	tenant: string;
-	clientId: string;
-	clientSecret: string;
-	resource: string;
-}
-
 export class Auth {
 	context: ExtensionContext;
-	config = workspace.getConfiguration('appConfig');
-
-	appConfig = this.config.get<AppConfig>('appConfig');
-	authorityUrl = `${this.appConfig.instance}/${this.appConfig.tenant}`;
+	config = workspace.getConfiguration('markdown');
+	instance = 'https://login.microsoftonline.com';
+	tenant = this.config.get<string>('tenant');
+	clientId = this.config.get<string>('clientId');
+	clientSecret = this.config.get<string>('clientSecret');
+	resource = 'https://graph.microsoft.com';
+	authorityUrl = `${this.instance}/${this.tenant}`;
 	constructor(context) {
 		this.context = context;
 	}
 	getToken = async (): Promise<TokenResponse> => {
 		const token: TokenResponse = this.context.globalState.get('token');
-		const expiresTime = new Date().getTime() / 1000;
-		if (!token || token.expiresIn < expiresTime) {
+		const expiresTime = Math.round(new Date().getTime() / 1000);
+		if (
+			!token ||
+			(token.expiresIn &&
+				Math.round((new Date().getTime() + token.expiresIn * 1000) / 1000) < expiresTime)
+		) {
 			const authContext = new AuthenticationContext(this.authorityUrl);
 			return new Promise((resolve, reject) => {
 				authContext.acquireTokenWithClientCredentials(
```

Comment:
Here you should get the access token of our AAD application, not the Microsoft Graph, so you need change the **resource** to our application id url:

Sandbox for testing: api://18c498b3-2ced-4ea0-b6ea-c52841b603de
PROD: api://79abd399-c39f-4589-8dda-ce6f7d6f8d70

I referenced from this page: https://docs.microsoft.com/en-us/javascript/api/overview/azure/activedirectory?view=azure-node-latest

![image](https://user-images.githubusercontent.com/19990166/85489491-a608b600-b602-11ea-888e-5939ee885a9a.png)

Reply:
Example code: https://github.com/AzureAD/azure-activedirectory-library-for-nodejs/blob/master/sample/client-credentials-sample.js

Reply:
@OsmondJiang Thank you for that. Once I switched the resouce to the api:// url the authorization to the allowlist started working.

Reply:
👍👍 great to see that, if you verified the Sandbox is working, then PROD should be OK.